File: src/main/GO/leetcode/lists/offer/Offer65.go
package offer

/**
 * @author wujingxinit@outlook.com
 * @date 2023/1/3 22:32
 */

func add(a int, b int) int {
	for b != 0 {
		carry := (a & b) << 1
		a = a ^ b
		b = carry
	}
	return a
}

File: java/src/leetcode/lists/offer/Offer65.java
package leetcode.lists.offer;

/**
 * @author wujingxinit@outlook.com
 * @date 2023/9/14 02:15
 */
public class Offer65 {
    public int add(int a, int b) {
        while (b != 0) {
            int carry = (a & b) << 1;
            a = a ^ b;
            b = carry;
        }
        return a;
    }
}
